### ISSUE TITLE
Enhance bgtask for GoTime

### ISSUE THREAD

**Issue by justenwalker** [gobridge/gopherbot#31]
Since the Twitter app has been de-authorized on Gophers Slack; updates from the GoTime Twitter Feed / Mastodon Feed will no longer work. The preferred way of pushing these updates to Slack is through Gopher Bot.

This issue is to propose adding this support to the existing bgtask for gotime.

## Proposal
1. Create a new package `internal/poller/gotimestatus`
2. Define NotifyFunc in `internal/poller/gotimestatus`. This function is called by `poller/gotimestatus.GoTimeStatus` whenever a new Mastodon post is discovered with the URL of the status.
  
    ```go
    type NotifyFunc func(ctx context.Context, statusURL string) error
    ```

3. Poll `@gotime@changelog.social` using the [Account/Statuses](https://docs.joinmastodon.org/methods/accounts/#statuses) API: `https://changelog.social/api/v1/accounts/109349735213354404/statuses?since_id=LAST_ID`
  - If there is no last status ID, post the latest status (only one)
  - If there is a last status ID
    - Get statuses since that ID  using `since_id`
    - Skip any statuses older than maxStatusAge. We only want to post fresh statuses, and this doubles as a way to avoid spamming the channel with the same update if redis persistence fails.
    - Call NotifyStatus for each statuses newer than the id.
    - Store the last status ID so it doesn't re-post again on the next call to Poll.
3. Create `cmd/bgtasks/gotimestatus.go` to set up the gotime status poller. 
4.  Define `goTimeStatusNotifyFactory` in `gotimestatus.go` that creates a function which uses the Slack Client to call [chat.postMessage](https://api.slack.com/methods/chat.postMessage) with `unfurl_links=true` and the statusURL as text. This allows Slack to do most of the heavy lifting reading the HTML from Mastodon and translating that into a Slack post. Example Below.

    <img width="658" alt="gotime_mastodon" src="https://user-images.githubusercontent.com/222027/204394746-8f7da53a-ad41-4496-8c61-871e2c5b63d4.png">

## Implementation Questions

1. Should the GopherBot post as itself (ie: @gopher) or should it customize the slack message to use a Profile Image + Name. We can use any image and name. If there is an emoji we can also use that as the picture. Example: 
<img width="651" alt="gotime_changelog" src="https://user-images.githubusercontent.com/222027/204442406-e4d90793-9613-45e7-914c-3339eec62ceb.png">

## Updates

Originally I had proposed to modify the gotime poller to add this functionality. But I did a little PoC and it is much cleaner to make a new package and copy/paste the implementation into the new package, making the necessary modifications. It gets real messy trying to shoehorn it into the existing poller.

**Comment by ardan-bkennedy** [comment]
@justenwalker this is awesome. @dlsniper can't review this until tomorrow, but this looks great to me. I'm sure Florin will have more insights and answers to your questions.

**Comment by dlsniper** [comment]
Thank you for your contribution!
I'll take a look at this and the related PR shortly.

**Comment by dlsniper** [comment]
Before I can understand what we need to do here, could you please help me understand:
- If the current implementation has a problem, what it is and why can't it be fixed?
- When is the status updated at `https://changelog.social/api/v1/accounts/109349735213354404/statuses`. The current implementation should post a start message at most a minute after a show starts. Will the new implementation do the same?

Thank you

**Comment by justenwalker** [comment]
> @dlsniper:
> If the current implementation has a problem, what it is and why can't it be fixed?

There is no current implementation. There are two distinct Go Time Pollers:
- `gotime` - (**Unchanged**) This current implementation announces when Go Time goes live.
- `gotimestatus` - (**NEW**) This replaces the Twitter integration. Whenever they make a `@gotime@changelog.social` post on Mastodon, that post should be available on the `#gotimefm` channel in Gopher Slack. These are not typically a "We're Live" announcement. They are usually "We have a new episode" or "Take this Unpopular Opinion Poll" and the results of said poll. See https://changelog.social/@gotime for examples.

> @dlsniper:
>  When is the status updated at `https://changelog.social/api/v1/accounts/109349735213354404/statuses`. The current implementation should post a start message at most a minute after a show starts. Will the new implementation do the same?

This URL is a Mastodon status feed for `@gotime@changelog.social`, not a "We're live" signal; So there is no current implementation. 

That said, if we want 1 minute latency for Mastodon posts as well, then I can update the PR to set the polling interval to 1 minute. We'll also want to update the max post age accordingly. I made some comments in the implementation PR #32 about the trade-offs. @jerodsanto might want to weigh in if this is too frequent. 5 Mins seems pretty good to me for social posts as they are not as time-sensitive as tuning in to the stream when It's Go Time.

**Comment by jerodsanto** [comment]
I'm cool with either 5 minutes or 1 minute. Your call!

**Comment by dlsniper** [comment]
@justenwalker thank you for the details! I'll switch to the PR.

**Comment by justenwalker** [comment]
Addressed by #32
